docs: combine monitor without helm content (#11293)
grafana/mimir: docs/sources/mimir/manage/monitor-grafana-mimir/monitor-system-health.md

# Monitor the health of your system

> To monitor and understand the health of your Grafana Mimir or Grafana Enterprise Metrics (GEM) database, you should collect and store the metrics and logs it exposes. This process is called _metamonitoring_. You're "monitoring" your "monitoring database."

Reviewer: ````suggestion
To monitor and understand the health of your Grafana Mimir or Grafana Enterprise Metrics (GEM) database, you should collect and store the metrics and logs it exposes. This process is called _meta-monitoring_. You're "monitoring" your "monitoring database."
````

## Collect metrics and logs from Grafana Mimir

{{< admonition type="note" >}}

> The Grafana Mimir Helm chart contains built-in configurations for metamonitoirng that use the Grafana Agent, which is now deprecated. Therefore, we no longer recommend using this approach. Instead we recommend an approach that uses the [Kubernetes Monitoring Helm chart](https://github.com/grafana/k8s-monitoring-helm/tree/main/charts/k8s-monitoring).

Reviewer: ````suggestion
The Grafana Mimir Helm chart contains built-in configurations for meta-monitoring that use the Grafana Agent, which is now **deprecated**. Therefore, we no longer recommend using this approach. Instead we recommend an approach that uses the [Kubernetes Monitoring Helm chart](https://github.com/grafana/k8s-monitoring-helm/tree/main/charts/k8s-monitoring).
````

{{< /admonition >}}

> To collect metrics and logs from Grafana Mimir, use the [built-in Mimir integration](https://github.com/grafana/k8s-monitoring-helm/blob/main/charts/k8s-monitoring/charts/feature-integrations/docs/integrations/mimir.md) provided by the [grafana/k8s-monitoring-helm](https://github.com/grafana/k8s-monitoring-helm) chart. This configures [Grafana Alloy](https://grafana.com/docs/alloy/latest/) to handle all scraping and log collection automatically—no `ServiceMonitors` are needed. You should not enable any metaMonitoring settings in the Mimir Helm chart (see note). 

Reviewer: ````suggestion
To collect metrics, logs and traces from Grafana Mimir, use the [built-in Mimir integration](https://github.com/grafana/k8s-monitoring-helm/blob/main/charts/k8s-monitoring/charts/feature-integrations/docs/integrations/mimir.md) provided by the [grafana/k8s-monitoring-helm](https://github.com/grafana/k8s-monitoring-helm) chart. This configures [Grafana Alloy](https://grafana.com/docs/alloy/latest/) to handle all scraping and log collection automatically, no `ServiceMonitors` are needed. You should not enable any `metaMonitoring` settings in the Mimir Helm chart (see note).
```

> See the [meta-monitoring example](https://github.com/grafana/k8s-monitoring-helm/tree/main/charts/k8s-monitoring/docs/examples/meta-monitoring) as a reference. You will need to update the `destinations` section to the metrics and logs databases that you'd like to use to store the metrics and logs scraped from Grafana Mimir or GEM. Metrics must be stored in Prometheus or a Prometheus-compatible TSDB (e.g., Thanos, a separate Grafana Mimir, or Grafana Cloud Metrics). You can even define multiple metrics and/or logs `destinations` to allow you to forward the metrics and logs to multiple places. 

Reviewer: ````suggestion
Refer to the [meta-monitoring example](https://github.com/grafana/k8s-monitoring-helm/tree/main/charts/k8s-monitoring/docs/examples/meta-monitoring) for guidance. Update the [`destinations`](https://github.com/grafana/k8s-monitoring-helm/tree/main/charts/k8s-monitoring/docs/destinations) section to specify where to send the metrics, logs, and traces collected from Grafana Mimir or GEM. Metrics must be sent to Prometheus or a Prometheus-compatible TSDB (such as Thanos, another Mimir instance, or Grafana Cloud Metrics). You can configure multiple metrics and/or logs destinations to forward data to several backends simultaneously.
````

## Visualize metrics and logs from Grafana Mimir

> Once you have collected the metrics and logs collected from your Grafana Mimir or GEM and stored them in separate metrics and logs databases, you can create dashboards and alerts from this data.

Reviewer: ````suggestion
Once you've collected metrics, logs, and traces from Grafana Mimir or GEM and stored them in their respective databases, you can use this data to build dashboards and define alerts.
````

> Start by configuring Grafana to read from these metrics and logs databases. This requires adding Prometheus and Loki datasources that point to these databases. 

Reviewer: ````suggestion
Start by configuring Grafana with datasources for metrics, logs, and traces. This typically involves adding Prometheus, Loki, and Tempo datasources that point to the respective backends.
````

As part of _metamonitoring_, you can create dashboards and receive alerts about the metrics and logs collected from Mimir. To set up these dashboards and alerts,
refer to [Installing Grafana Mimir dashboards and alerts](https://grafana.com/docs/mimir/<MIMIR_VERSION>/manage/monitor-grafana-mimir/installing-dashboards-and-alerts/).

> ## Configure the Grafana Agent operator through the Helm chart

Reviewer 2: ```suggestion
## Monitoring Grafana Mimir when setting up without the Helm chart
```

Author: Updated to `Monitor Grafana Mimir when setting up with the Helm chart`, since that's what this section describes.

          passwordSecretKey: metamonitoring
```

> ## Install Grafana Mimir without the Helm chart

Author: Updated to `Monitor Grafana Mimir when setting up without the Helm chart` to use the infinitive tense.

It's possible that you need to modify this configuration. For
more information, see [dashboards and alerts requirements](../requirements/).

### Service discovery

As a best practice, deploy Grafana Mimir on Kubernetes. The Grafana Alloy configuration relies on Kubernetes service discovery and Pod labels to constrain the collected metrics and
logs to ones that are strictly related to the Grafana Mimir deployment. If you are deploying Grafana Mimir on something other than Kubernetes, then replace the `discovery.kubernetes` component with another [Alloy component](https://grafana.com/docs/alloy/latest/reference/components) that can discover the Mimir processes.

### Collect metrics and logs via Grafana Alloy

Set up Grafana Alloy to collect logs and metrics from Mimir or GEM. To get started with Grafana Alloy,
refer to [Get started with Grafana Alloy](https://grafana.com/docs/alloy/<ALLOY_VERSION>/get-started). After deploying Alloy, refer to [Collect and forward Prometheus metrics](https://grafana.com/docs/alloy/<ALLOY_VERSION>/collect/prometheus-metrics/) for instructions on how to configure your Alloy instance to scrape Mimir or GEM.

### Monitor without the Helm chart

> To monitor the health of your system without using the Helm chart, see [Collect metrics and logs without the Helm chart](https://grafana.com/docs/mimir/<MIMIR_VERSION>/manage/monitor-grafana-mimir/collecting-metrics-and-logs/#collect-metrics-and-logs-without-the-helm-chart).

Author: Good catch! Removed.
